index.js

```diff
@@ -69,16 +72,40 @@ module.exports = function (options = {}) {
 				name += `/${parts.shift()}`;
 			}
 
-			if (parts.length > 0) return;
+			const entry = parts.join('/') || '.';
+
+			let pkg;
+			let dir;
 
 			let search_dir = importer;
 			while (search_dir !== (search_dir = path.dirname(search_dir))) {
-				const dir = path.join(search_dir, 'node_modules', name);
+				dir = path.join(search_dir, 'node_modules', name);
 				const file = path.join(dir, 'package.json');
 				if (fs.existsSync(file)) {
-					const pkg = JSON.parse(fs.readFileSync(file, 'utf-8'));
-					if (pkg.svelte) {
-						return path.resolve(dir, pkg.svelte);
+					pkg = JSON.parse(fs.readFileSync(file, 'utf-8'));
+					break;
+				}
+			}
+
+			if (pkg) {
+				// resolve pkg.svelte first
+				if (entry === '.' && pkg.svelte) {
+					return path.resolve(dir, pkg.svelte);
+				}
+
+				const resolved = await this.resolve(importee, importer, { skipSelf: true });
+
+				// if we can't resolve this import without the `svelte` condition, warn the user
+				if (!resolved) {
+					try {
+						resolve(pkg, entry, { conditions: ['svelte'] });
+
+						if (!warned) {
+							console.error('\n\u001B[1m\u001B[31mWARNING: Your @rollup/plugin-node-resolve configuration\'s \'exportConditions\' array should include \'svelte\'. See https://github.com/sveltejs/rollup-plugin-svelte#svelte-exports-condition for more information\u001B[39m\u001B[22m\n');
```

Comment:
Should we first say what it was that we failed to resolve? It might be a better message in the case that resolving fails for any other reason. I don't have any other reason in mind, but it might protect against something unexpected happening

Reply:
I think the message is fine as is, I don't see a situation where it would be wrong.